Fix language in section 01 01 01

gchang12 opened the pull request:
Redid the edits introduced in https://github.com/java/devjava-content/pull/178#issuecomment-3447412420

Corrected grammar and made a few more edits, including but not limited to:
- replacing 'expand' with 'extract' where appropriate.
- enclosing function names within backticks.
- inserting commas where appropriate
- inserting key-names inside angle-brackets

TODO: Inspect for missing resources marked by `<!-- NOTE:`

gchang12 committed:
Recreated changes introduced in now-invalid PR.

https://github.com/java/devjava-content/pull/178#issuecomment-3447412420

gchang12 committed:
Replaced instances of 'expand' with 'extract'; added commas and touched up language

gchang12 committed:
Enclosed key names inside tags, fixed language, and pointed out possibly missing resources.

gchang12 committed:
Minor fixes

danthe1st commented on app/pages/learn/01_tutorial/01_your-first-java-app/01_getting-started-with-java.md:
```diff
@@ -40,28 +40,28 @@ There are several steps that you need to follow to create a Java application. Th
 > These two steps: compilation and execution require two specific pieces of software that are part of the Java Development Kit, also known as the JDK. You will see how to download the JDK for free and how to install it later in this tutorial.
 
-Note that starting with Java SE 11 you can also merge these two steps into one, by executing a `.java` file directly. You can use these feature only if you are executing a program that is written in a single file. This way of executing your java application does not work if your java code spans more than one file.
+Note that starting with Java SE 11 you can also merge these two steps into one, by executing a `.java` file directly. You can use these feature _only_ if you are executing a program that is written in a single file.
```
Technically that's outdated with JDK 22 (which [allows running multi-file programs like that](https://openjdk.org/jeps/458)) but I am not sure about the best way to say that.

gchang12 replied:
How's this?

https://github.com/java/devjava-content/pull/179/commits/df3fbc274cf4e38fe09044b08f997a199480aee6

danthe1st commented on app/pages/learn/01_tutorial/01_your-first-java-app/01_getting-started-with-java.md:
```diff
@@ -74,14 +74,14 @@ You can give this class any name as long as it does not start with a number. The
 If you are following this example to the letter, you should save the `MyFirstClass` class in a text file called `MyFirstClass.java`.
 
-Just a word of warning: you should be using a plain text editor to create and save this file. Using a word processor will not work.
+Just a word of warning: you should be using a plain text editor to create and save this file. Using a word processor like Microsoft Word or Google Docs will not work.
```
I am personally not sure whether including this yields any benefit but I personally don't care about that to be honest.

gchang12 replied:
Change has been reverted in https://github.com/java/devjava-content/pull/179/commits/bdb4ad7f4982f2eb37d1619fe00b6c08d1eef6c8

danthe1st commented on app/pages/learn/01_tutorial/01_your-first-java-app/01_getting-started-with-java.md:
```diff
@@ -94,7 +94,7 @@ You may also have heard about J2EE, Java EE or Jakarta EE. All these acronyms re
 ## Setting up a Java Development Kit
 
-You can download the JDK from different places. There is a one-stop page that always refers to the latest version of the JDK: https://jdk.java.net/. Selecting the latest "Ready for use" JDK version takes you to a page where you can download the version of the JDK you need.
+You can download the JDK from different places. There is a one-stop page that always refers to the latest version of the JDK: https://jdk.java.net/. Selecting a JDK version listed under the label, "Ready for use" redirects you to a page where you can download the version of the JDK you need.
```
Are you sure about the comma after "label"?

gchang12 replied:
Erased comma in https://github.com/java/devjava-content/commit/bdb4ad7f4982f2eb37d1619fe00b6c08d1eef6c8#diff-e57ee80810caa9db25eae7bfaebfaf55d2daebb09652c58eded37c317a637814R97

danthe1st commented on app/pages/learn/01_tutorial/01_your-first-java-app/01_getting-started-with-java.md:
```diff
@@ -259,39 +259,39 @@ Your shell should print the complete path to the `java` executable file in the `
 ### Compiling and Running Your First Java Program
 
-Whether you followed the Windows, the Linux or the macOS path, the remaining is the same.
+The following instructions are the same regardless of if you followed the the Windows, the Linux or the macOS path.
```
You have a duplicate "the" here.

Apart from that, I would personally prefer "whether" over "if" here but that's just a preference.

gchang12 replied:
Removed duplicate 'the' in https://github.com/java/devjava-content/pull/179/commits/bdb4ad7f4982f2eb37d1619fe00b6c08d1eef6c8

'whether' is typically used for one of two choices.
Source: Definition 2 of https://www.merriam-webster.com/dictionary/whether

danthe1st commented on app/pages/learn/01_tutorial/01_your-first-java-app/01_getting-started-with-java.md:
```diff
@@ -259,39 +259,39 @@ Your shell should print the complete path to the `java` executable file in the `
 
-1. Change to the directory where you saved your first class `MyFirstClass.java`. You can check that you are in the right directory by typing `dir`. It will show you the files you have in this directory. You should see your `MyFirstClass.java` file.
-2. Check that your compiler is accessible from this directory by typing the following. This command is the same whether you are on Windows or Linux.
+1. Change to the directory where you saved your first class, `MyFirstClass.java`. You can check that you are in the right directory by typing `dir` (for Windows) `ls` (for macOS or Linux). This command will list the files you have in this directory. You should see your `MyFirstClass.java` file in that list.
```
I think there should be some sort of separation/connector between `dir` and `ls`, for example
```
You can check that you are in the right directory by typing `dir` (on Windows) or `ls` (for macOS or Linux).
```

gchang12 replied:
Good call.

gchang12 replied:
Added conjunction in https://github.com/java/devjava-content/pull/179/commits/bdb4ad7f4982f2eb37d1619fe00b6c08d1eef6c8

danthe1st commented on app/pages/learn/01_tutorial/01_your-first-java-app/01_getting-started-with-java.md:
```diff
@@ -259,39 +259,39 @@ Your shell should print the complete path to the `java` executable file in the `
-
-If the compiler remains silent and does not complain about anything: congratulations! It means that your Java code has been properly compiled. Checking the content of the directory again should show a new file in it: `MyFirstClass.class`
+Two things may happen at this point.
+- You will have error messages telling you that the compiler cannot compile your code, because of errors in your Java code. You will need to fix them before being able to move on.
```
I am unsure about the word "will" (in "You will have error messages"). This statement is one possibility so saying the user "will" have an error message seems inaccurate (as opposed to the "may have" in the original version).
The same also applies for the next line.

danthe1st commented on app/pages/learn/01_tutorial/01_your-first-java-app/01_getting-started-with-java.md:
```diff
@@ -259,39 +259,39 @@ Your shell should print the complete path to the `java` executable file in the `
 ## Adding Code to Your Class to Run it
 
-So far your class is empty; there is no executable code in it. If you were able to compile it properly then you can advance to the next step and execute some code.
+So far your class is empty; there is no executable code in it. If you were able to compile it properly, then advance to the next line of instruction to execute some code:
```
This might just be personal preference but that change feels a bit weird to me - both the `, then` (the "then" seems to be redundant) and saying "next line of instruction".

danthe1st commented on app/pages/learn/01_tutorial/01_your-first-java-app/01_getting-started-with-java.md:
```diff
@@ -302,30 +302,32 @@ public class MyFirstClass {
+There is technical code in this class that may not be very clear to you right now. Don't worry; all you need to do is to compile it by following the steps described in the previous section.
 
-Make sure that the compiler created the `MyFirstClass.class` for you. To run it, all you need to type is the following command:
+Make sure that the compiler has created the `MyFirstClass.class` for you. To run it, type is the following command:
```
The "is" in "To run it, type _is_ the following command" doesn't belong here.

danthe1st commented on app/pages/learn/01_tutorial/01_your-first-java-app/01_getting-started-with-java.md:
````diff
@@ -302,30 +302,32 @@ public class MyFirstClass {
 ```
 
-This should print _Hello, World!_ on the console. If this is the case: congratulations! You have been able to run your first Java program!
+This should print _Hello, World!_ on the console. If this is the case, then congratulations! You were able to run your first Java program!
````
Is ", then congratulations" a common thing to say? At least I'm not used to it.

danthe1st commented on app/pages/learn/01_tutorial/01_your-first-java-app/01_getting-started-with-java.md:
````diff
@@ -340,19 +342,19 @@ javac is not recognized as an internal or external command, operable program or
 ```
 
-If you choose this option, you'll have to precede your `javac` and `java` commands with `C:\jdk19\bin\` each time you compile or run a program. To avoid this extra typing, consult the section Updating the `PATH` variable in the JDK 19 installation instructions.
+If you choose this option, you'll have to precede your `javac` and `java` commands with `C:\jdk19\bin\` each time you compile or run a program. To avoid this extra typing, consult the section [Updating the `PATH` variable](<!-- NOTE: Missing hyperlink -->) in the JDK 19 installation instructions.
````
I think the `<!-- NOTE: Missing hyperlink -->` is here by mistake.

gchang12 replied:
That was a placeholder. I meant to put in the hyperlink to that article, but I wasn't able to find it.

gchang12 replied:
I think the change introduced in this commit should fix this.

https://github.com/java/devjava-content/pull/179/commits/c62bb7f2c95494e7c1b78b3c10df4f0b66f9567e

I also removed the '19' to avoid the need to change this in the future.

danthe1st commented on app/pages/learn/01_tutorial/01_your-first-java-app/01_getting-started-with-java.md:
```diff
@@ -409,23 +416,25 @@ Again, your program did not successfully compile, and the compiler did not creat
 
 #### Error Messages on Microsoft Windows Systems
 
+<!-- NOTE: What error? -->
```
This comment seems to be a leftover that should be addressed before merging as well.

gchang12 replied:
Further below, on line 421, there's this paragraph:

> If you receive this error, `java` cannot find your bytecode file, `HelloWorldApp.class`.

There's no previous codeblock containing the error in question.

danthe1st commented on app/pages/learn/01_tutorial/01_your-first-java-app/01_getting-started-with-java.md:
````diff
@@ -409,23 +416,25 @@ Again, your program did not successfully compile, and the compiler did not creat
 ```
 
-The prompt should change to `C:\java>`. If you enter `dir` at the prompt, you should see your `.java` and `.class` files. Now enter `java HelloWorldApp` again.
+The prompt should change to `C:\java>`. If you enter `dir` at the prompt, you should see your `.java` and `.class` files. Now enter `java HelloWorldApp` and press \<Return\> again.
````
Question: Would it be better to be consistent between "Enter" and "Return"?

gchang12 replied:
You have a point. The original docs already had 'Enter', so I should have stuck with that. My mistake.

gchang12 replied:
Replaced as of https://github.com/java/devjava-content/pull/179/commits/bdb4ad7f4982f2eb37d1619fe00b6c08d1eef6c8

danthe1st commented on app/pages/learn/01_tutorial/01_your-first-java-app/01_getting-started-with-java.md:
```diff
@@ -485,9 +494,9 @@ This first Java program showed you the basic steps every Java developer follows
+Developers that work on large applications do not use plain text editors to manage their source code; they use Integrated Development Environments, or "IDEs". IDEs are complex software applications, specialized in software development. These applications handle the compilation of your source code automatically, help you track errors in the syntax of your Java code and nail down bugs in its execut ...
 
-Some of these tools are open source and free to use.
+Some of these tools are open-source and free to use.
```
I typically write it as "Open Source" (without a dash) and it seems [the OSI does so as well](https://opensource.org/osd). Though whether/when it should be capitalized is another question I don't have a good answer to.

gchang12 replied:
Reverted in https://github.com/java/devjava-content/pull/179/files/8e1709cbf46b056f6344439dcf57513d3fdbdfe1#diff-e57ee80810caa9db25eae7bfaebfaf55d2daebb09652c58eded37c317a637814

danthe1st reviewed (changes requested):
Please note that I am not a maintainer and I'm not working at Oracle. I don't have any authority here. What I'm writing here are merely suggestions and what you need to change is to be decided by the maintainers.

That being said, this PR might be a good chance to also change the article to install/use JDK 25 instead of JDK 19 (As Oracle designated JDK 25 to be an LTS, that's also what people would be more likely to use I guess).

gchang12 committed:
Update 01_getting-started-with-java.md

Incorporated suggestions from review by danthe1st

gchang12 committed:
Update 01_getting-started-with-java.md

Mentioned multi-file program execution-from-command-line feature introduced in JDK 22.

gchang12 committed:
Update 01_getting-started-with-java.md

Inserted hyperlink to article explaining how to update PATH variable
